Type 'Module | null | undefined' is not assignable to type 'NodeModule | null'

The latest `@types/webpack-env@1.15.3` no longer works together with the latest node v12 types `@types/node@12.19.2`.

Error:
```
node_modules/@types/webpack-env/index.d.ts:349:15 - error TS2430: Interface 'NodeJS.Module' incorrectly extends interface '__WebpackModuleApi.Module'.
  Types of property 'parent' are incompatible.
    Type 'Module | null | undefined' is not assignable to type 'NodeModule | null | undefined'.
      Type 'Module' is not assignable to type 'NodeModule'.
        Types of property 'parent' are incompatible.
          Type 'Module | null | undefined' is not assignable to type 'NodeModule | null'.
            Type 'undefined' is not assignable to type 'NodeModule | null'.

349     interface Module extends __WebpackModuleApi.Module {}
```

It seems it's not a problem in the `@types/webpack-env` package itself, but in the `@types/node` package.

[This change](https://github.com/Semigradsky/DefinitelyTyped/commit/aec9a1a0ba4412aa1d5faed875280f0a467e31a6#diff-fbb362680b965a40c12d1baf1c1380c6963ce86b2106ca8fafaba7323d91c009R1186) seems to have caused the error.

The changed `Module` [class](https://github.com/Semigradsky/DefinitelyTyped/blob/aec9a1a0ba4412aa1d5faed875280f0a467e31a6/types/node/v12/globals.d.ts#L1167) is no longer compatible to the `NodeModule` interface (in the same file). This is because the `parent` [property](https://github.com/Semigradsky/DefinitelyTyped/blob/aec9a1a0ba4412aa1d5faed875280f0a467e31a6/types/node/v12/globals.d.ts#L234) of the `NodeModule` interface has not been adjusted to match the changed one from the `Module` class.

The `@types/webpack-env` package just uses the `NodeModule` interface [here](https://github.com/Semigradsky/DefinitelyTyped/blob/aec9a1a0ba4412aa1d5faed875280f0a467e31a6/types/webpack-env/index.d.ts#L71) and expects it to be compatible to the `Module` class (which was the case before the change) [here](https://github.com/Semigradsky/DefinitelyTyped/blob/aec9a1a0ba4412aa1d5faed875280f0a467e31a6/types/webpack-env/index.d.ts#L349).

There was a [similar issue](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/45927) in the past (just not the `parent` property but a different one).

@Semigradsky @use-strict @rhonsby

... basically said, this code compiled before the [change](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/48570/files#diff-fbb362680b965a40c12d1baf1c1380c6963ce86b2106ca8fafaba7323d91c009R1186) to the `Module` class:
```typescript
/// <reference types="node" />
const moduleClass: NodeJS.Module = {} as any;
const moduleInterface: NodeModule = moduleClass;
```
while with `@types/node@12.19.2` this throws:
```
index.ts:3:7 - error TS2322: Type 'Module' is not assignable to type 'NodeModule'.
  Types of property 'parent' are incompatible.
    Type 'Module | null | undefined' is not assignable to type 'NodeModule | null'.
      Type 'undefined' is not assignable to type 'NodeModule | null'.

3 const moduleInterface: NodeModule = moduleClass;
```

--- a/types/node/v12/globals.d.ts
+++ b/types/node/v12/globals.d.ts
@@ -231,7 +231,8 @@ interface NodeModule {
     id: string;
     filename: string;
     loaded: boolean;
-    parent: NodeModule | null;
+    /** @deprecated since 12.19.0 Please use `require.main` and `module.children` instead. */
+    parent: NodeModule | null | undefined;
     children: NodeModule[];
     /**
      * @since 11.14.0

--- a/types/node/v12/test/global.ts
+++ b/types/node/v12/test/global.ts
@@ -7,6 +7,7 @@ import { Readable, Writable } from 'stream';
     x.parent = require.main!;
     require.main = y;
     x.path; // $ExpectType string
+    y.parent; // $ExpectType NodeModule | null | undefined
 }
 
 {

